Kickstarter-iOS/ViewModels/AppDelegateViewModel.swift

```diff
@@ -1188,3 +1210,38 @@ private func configRetainingDebugFeatureFlags(_ config: Config) -> Config {
 
   return config |> Config.lens.features .~ currentFeatures.withAllValuesFrom(storedFeatures)
 }
+
+private func willHandleBrazeUrlAndNavigation(urlAndExtras: (URL?, [AnyHashable: Any]?)?)
+  -> (Bool, Navigation?) {
+  guard
+    let (maybeUrl, _) = urlAndExtras,
+    let url = maybeUrl
+  else { return (false, nil) }
+
+  guard let navigation = Navigation.match(url) else {
+    return (false, nil)
+  }
+
+  return (true, navigation)
+}
+
+private func updateUserNotificationSetting(navigation: Navigation) -> SignalProducer<User, Never> {
+  guard
+    case let .settings(.notifications(notification, enabled)) = navigation,
+    let currentUser = AppEnvironment.current.currentUser
+  else { return .empty }
+
+  let currentNotifications = AppEnvironment.current.currentUser?.notifications.encode()
+  let updatedNotifications = currentNotifications?.withAllValuesFrom([notification: enabled])
+
+  guard
+    let data = try? JSONSerialization.data(withJSONObject: updatedNotifications as Any, options: []),
+    let userNotifications = try? JSONDecoder().decode(User.Notifications.self, from: data)
+  else { return .empty }
+
+  let updatedUser = currentUser |> User.lens.notifications .~ userNotifications
+
+  return AppEnvironment.current.apiService.updateUserSelf(updatedUser)
+    .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+    .demoteErrors()
```

Comment:
We're not handling errors here so we just allow this to fail silently if it does.